README.md
## [fhir-ui](https://healthintellect.com)

[![npm package](https://img.shields.io/npm/v/fhir-ui?style=flat-square)](https://www.npmjs.org/package/fhir-ui)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/react?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/react-dom?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/@material-ui/core?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/@material-ui/pickers?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/@material-ui/icons?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/moment?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/@date-io/core?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/@date-io/moment?style=flat-square)
![npm peer dependency version](https://img.shields.io/npm/dependency-version/fhir-ui/peer/clsx?style=flat-square)

fhir-ui [React](http://facebook.github.io/react/) components are used in conjunction with the [Material UI](http://www.material-ui.com/) component library to display [HL7 FHIR Resources](https://www.hl7.org/fhir/resourcelist.html) using [Google's Material Design](https://www.google.com/design/spec/material-design/introduction.html) specification.

## Prerequisites

- [React - Component Rendering](http://facebook.github.io/react/)
- [Material-UI - User Interface](http://material-ui.com/#/get-started/prerequisites)
- [Material-UI - Icons](https://material-ui.com/components/material-icons/)
- [Material-UI - Pickers](https://material-ui.com/components/pickers/)
- [Moment.js](https://momentjs.com/)
- [Fast Healthcare Interoperatbility Resources](https://www.hl7.org/fhir/resourcelist.html)
- [Roboto Font](http://www.google.com/fonts#UsePlace:use/Collection:Roboto:400,300,500)

## Installation

fhir-ui is available as an [npm package](https://www.npmjs.org/package/fhir-ui).

```sh
npm install --save fhir-ui
```

or

```sh
yarn add fhir-ui
```

## Theming

fhir-ui components require a theme to be provided. The quickest way to get up and running is by using the Material-UI `ThemeProvider` to inject the theme into your application context. Following that, you can to use any of the components as demonstrated in the documentation. Here is a quick example to get you started:

```jsx
import React from 'react'
import ReactDOM from 'react-dom'
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles'
import { MuiPickersUtilsProvider } from '@material-ui/pickers'
import lightBlue from '@material-ui/core/colors/lightBlue'

import { PatientCard } from 'fhir-ui'

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    primary: {
      light: lightBlue[300],
      main: lightBlue[500],
      dark: lightBlue[900],
      contrastText: '#FFF',
    },
    secondary: {
      light: lightBlue[300],
      main: lightBlue[500],
      dark: lightBlue[900],
      contrastText: '#FFF',
    },
    background: {
      default: '#f9f9f9',
    },
  },
})

const App = () => (
  <ThemeProvider theme={theme}>
    <MuiPickersUtilsProvider utils={MomentUtils}>
      <PatientCard patient={patient} />
    </MuiPickersUtilsProvider>
  </ThemeProvider>
)

ReactDOM.render(<App />, document.getElementById('app'))
```

**Patient Table Example**

```jsx
import React from 'react'
import { PatientTable } from 'fhir-ui'

const MyFhirComponent = () => (
  <div>
    <PatientTable
      patients={patientList}
      onRowClick={patientId => console.log('Selected Patient ID: ', patientId)}
    />
    <hr />
    <PatientDetail
      patient={patient}
      patientId={patient.id}
      onDelete={patientId => console.log('Removed Patient ID: ', patientId)}
    />
  </div>
)

export default MyFhirComponent
```

Please refer to each component's CodeSandbox.io example for an example on how to use them.

## CodeSandbox.io Examples

- [PatientCard](https://codesandbox.io/s/fhir-ui-patient-card-demo-yq5yg)
- [PatientTable](https://codesandbox.io/s/fhir-ui-patient-table-demo-4dbfx)
- [PatientDetail](https://codesandbox.io/s/fhir-ui-patient-detail-demo-9jhx6)
- [PatientBanner](https://codesandbox.io/s/fhir-ui-patient-banner-demo-1u9lm)
- [ObservationTable](https://codesandbox.io/s/fhir-ui-observation-table-demo-7nn4q)
- [ObservationDetail](https://codesandbox.io/s/fhir-ui-observation-detail-demo-2qcio)
- [ConditionTable](https://codesandbox.io/s/fhir-ui-condition-table-demo-7x4gm)
- [AllergyTable](https://codesandbox.io/s/fhir-ui-allergy-table-demo-g2kgo)
- [MedicationTable](https://codesandbox.io/s/fhir-ui-medication-table-demo-hxrt9)

## License

This project is licensed under the terms of the
[MIT license](https://github.com/healthintellect/fhir-ui/blob/master/LICENSE)


CHANGELOG.md
## fhir-ui Changelog

All notable changes to this project will be documented in this file.

### [v0.6.3][2022-06-23]

- Fix allergy table re-render on allergies change
- Fix condition table re-render on conditions change
- Fix medication table re-render on medications change
- Fix observation table re-render on observations change
- Fix observation detail re-render on observation change
- Fix patient card re-render on patient change
- Fix patient banner re-render on patient change
### [v0.5.7][2020-10-13]

- Fix full name on patient card
- Update CodeSandbox.io links in README.md
- Add GitHub package details to package.json

### [v0.5.5][2020-01-13]

- Fix error when empty patient passed to PatientBanner

### [v0.5.4][2020-01-06]

- Add tableHeight options to table components
- Add scrollbar to table components
- Fix CodeText column heading on AllergyTable
- Change tableSize to tableRowSize for all table components

### [v0.5.0][2020-01-05]

- Add MedicationTable component and documentation
- Add AllergyTable component and documentation
- Add ConditionTable component and documentation
- Modify table title location on PaientTable
- Added defaultRowsPerPage property to PatientTable
- Added example FHIR patient bundle

### [v0.4.2][2019-12-31]

- Add PatientBanner component and documentation

### [v0.3.8][2019-12-29]

- Fix date of birth field variant in PatientCard and PatientDetail

### [v0.3.7][2019-12-29]

- Fix documentation deployment to github pages
- Move docz to devdependency
- Update package.json keywords
- Add NPM badges in README.md
- Update change log

### [v0.3.6][2019-12-28]

- Resolve controlled component warnings in ObservationDetail
- Cleanup component properties
- Recreated examples in docz documentation
- Restructure directories
- Add docz documentation

### [v0.2.10] - 2019-12-25

- Add observation detail component

### [v0.2.7] - 2019-12-21

- Refactor to use React hooks
- Refactor to use `@material-ui/icons`
- Update npm modules

[unreleased]: https://github.com/healthintellect/fhir-ui/compare/v0.3.7...HEAD
[v0.3.6]: https://github.com/healthintellect/fhir-ui/compare/v0.3.6...v0.3.7
[v0.3.6]: https://github.com/healthintellect/fhir-ui/compare/v0.2.10...v0.3.6
[v0.2.10]: https://github.com/healthintellect/fhir-ui/compare/v0.2.9...v0.2.10
[v0.2.9]: https://github.com/healthintellect/fhir-ui/compare/v0.2.8...v0.2.9
[v0.2.8]: https://github.com/healthintellect/fhir-ui/compare/v0.2.7...v0.2.8
[v0.2.7]: https://github.com/healthintellect/fhir-ui/compare/v0.1.1...v0.2.7


example/src/App.js
import React from 'react'
import { Container, Grid, Typography, Link } from '@material-ui/core'
import { MuiPickersUtilsProvider } from '@material-ui/pickers'
import {
  createMuiTheme,
  makeStyles,
  ThemeProvider,
} from '@material-ui/core/styles'
import lightBlue from '@material-ui/core/colors/lightBlue'
import MomentUtils from '@date-io/moment'
import {
  PatientCard,
  PatientDetail,
  PatientTable,
  ObservationTable,
  ObservationDetail,
} from 'fhir-ui'

import {
  patient,
  patientTest,
  patients,
  observation,
  observations,
} from './fhir-examples'

const useStyles = makeStyles(theme => ({
  root: {
    margin: theme.spacing(6, 0, 3),
  },
  header: {
    margin: theme.spacing(1),
  },
  body: {
    display: 'flex',
    justifyContent: 'center',
    margin: theme.spacing(1),
  },
  footer: {
    margin: theme.spacing(1),
  },
}))

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    primary: {
      light: lightBlue[300],
      main: lightBlue[500],
      dark: lightBlue[900],
      contrastText: '#FFF',
    },
    secondary: {
      light: lightBlue[300],
      main: lightBlue[500],
      dark: lightBlue[900],
      contrastText: '#FFF',
    },
    background: {
      default: '#f9f9f9',
    },
  },
})

const Copyright = () => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright © '}
      <Link color="inherit" href="https://healthintellect.com/">
        Health Intellect
      </Link>
      {' 2019 - '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  )
}

const App = () => {
  const classes = useStyles()
  return (
    <ThemeProvider theme={theme}>
      <MuiPickersUtilsProvider utils={MomentUtils}>
        <Container>
          <div className={classes.header}>
            <Typography variant="h4" gutterBottom className={classes.root}>
              fhir-ui examples
            </Typography>
          </div>
          <div className={classes.body}>
            <Grid container spacing={3} direction="row">
              <Grid item xs={12} md={6}>
                <Typography variant="subtitle1" gutterBottom align="center">
                  Patient Card
                </Typography>
                <PatientCard patient={patientTest} fieldVariant="outlined" />
              </Grid>
              <Grid item xs={12} md={6}>
                <Typography variant="subtitle1" gutterBottom align="center">
                  Patient Detail
                </Typography>
                <PatientDetail patient={patientTest} fieldVariant="outlined" />
              </Grid>
              <Grid item xs={12}>
                <Typography variant="subtitle1" gutterBottom align="center">
                  Patient Table
                </Typography>
                <PatientTable patients={patients} />
              </Grid>
              <Grid item xs={12}>
                <Typography variant="subtitle1" gutterBottom align="center">
                  Observations Table
                </Typography>
                <ObservationTable
                  observations={observations}
                  hideActionIcons={true}
                  hideCheckboxes={true}
                  hideDevices={true}
                />
              </Grid>
              <Grid item xs={12}>
                <Typography variant="subtitle1" gutterBottom align="center">
                  Observations Detail
                </Typography>
                <ObservationDetail
                  observationId={observation.id}
                  observation={observation}
                  fieldVariant="outlined"
                />
              </Grid>
            </Grid>
          </div>
          <div className={classes.footer}>
            <Copyright />
          </div>
        </Container>
      </MuiPickersUtilsProvider>
    </ThemeProvider>
  )
}

export default App
